### PR TITLE
Feature/data share endpoint

### LINKED ISSUE #119: Create endpoint for accessing the "data share"

### PR DESCRIPTION
**Pull request by Jakedavies**
closes #119

### CODE REVIEW

**Comment by Jakedavies** [comment]
also closes #129  now

**Comment by sindelio** [comment]
Will review it first thing tomorrow!

File: cg/test/subjects/data-shares.api.spec.js [hunk 1]
```diff
@@ -131,3 +131,54 @@ describe('Data share remove', () => {
...
+    const token = await createUser(subjectId);
+
+    const res = await fetchWithAuthorization('/api/subject/data-shares/create', token, {
+      method: 'POST',
+      body: { name: 'no-decryption-key' }
+    });
+
+    expect(res.status).toEqual(200);
+    const [dataShare] = await db('data_shares').where({ name: 'no-decryption-key' });
+    console.log(dataShare);
```
Comment (maksymilian-majer) [inline cg/test/subjects/data-shares.api.spec.js]: Want to keep it here?

File: cg/src/domains/subjects/data-shares.service.js [hunk 2]
```diff
@@ -25,6 +31,26 @@ class DataShareService {
         id: dataShareId
       });
   }
+
+  async getDataForShare(token) {
+    const [data] = await db('data_shares')
+      .where({ token })
+      .join('subjects', 'data_shares.subject_id', 'subjects.id')
+      .leftJoin('subject_keys', 'subjects.id', 'subject_keys.subject_id')
```
Comment (sindelio) [inline cg/src/domains/subjects/data-shares.service.js]: Left Join means we want every entry that is on 'subjects' and also is is 'subject_keys' right?
If an entry is in 'subject_keys' but not in 'subjects' then it should be left out right

File: cg/src/domains/subjects/subjects.routes.js [hunk 3]
```diff
@@ -31,6 +35,12 @@ module.exports = app => {
     asyncHandler(processorsController.getProcessors.bind(processorsController))
   );
 
+  router.get(
+    '/data-shares/share',
```
Comment (sindelio) [inline cg/src/domains/subjects/subjects.routes.js]: So a subject can:
- Create a URL to share his data
- Remove the URL
- Share the URL

File: cg/test/subjects/data-shares.api.spec.js [hunk 4]
```diff
@@ -131,3 +131,54 @@ describe('Data share remove', () => {
...
+    const [dataShare] = await db('data_shares').where({ name: 'testdatasharing' });
+
+    const res2 = await fetch(`/api/subject/data-shares/share?token=${dataShare.token}`);
+
+    expect(res2.status).toEqual(200);
+    expect(await res2.json()).toEqual(expect.objectContaining({ customData: true }));
+  });
+
+  it('Should error if token is invalid', async () => {
+    const res = await fetch(`/api/subject/data-shares/share?token=badtoken`);
```
Comment (sindelio) [inline cg/test/subjects/data-shares.api.spec.js]: When no method is specified, it defaults to GET right?

File: cg/src/domains/subjects/subjects.routes.js (hunk 3 above)
Comment (Jakedavies) [reply cg/src/domains/subjects/subjects.routes.js]: correct

File: cg/test/subjects/data-shares.api.spec.js (hunk 4 above)
Comment (Jakedavies) [reply cg/test/subjects/data-shares.api.spec.js]: correct

**Pull request merged**